draft-trammell-taps-interface.md

```diff
@@ -984,6 +984,21 @@ The following values are valid for Transmission Profile:
   : This Message may be sent at the system's leisure. This can
   be used to signal a preference for less-than-best-effort treatment, to delay
   sending until lower-cost paths are available, and so on.
+  
+## Batching Sends {#send-batching}
+
+In order to reduce the overhead of sending multiple small Messages on a Connection, the
+application may want to batch several Send actions together. This provides a hint to
+the system that the sending of these Messages should be coalesced when possible,
+and that sending any of the batched Messages may be delayed until the last Message
+in the batch is enqueued.
+
+~~~
+Connection.Batch(
+	Connection.Send(Message, sendParameters)
+	Connection.Send(Message, sendParameters)
+)
```

Comment:
I'd almost leave out the pseudocode, but this is fine for now. Like events, the implementation platform is going to almost completely determine the syntax. In Go, I'd probably do something like:

```go
func (c Connection) Batch (sendfunc func());

var c Connection
c.Batch(func() {
    c.Send(...)
    c.Send(...)
})
```

which is almost the same thing but makes the closure explicit (I presume the Swift thing has a closure behind it anyway).

Another pattern I've seen (usually with RDBMS drivers and such) is to have a `Batch` object that has a `Send` method and a `Commit` method.

In C you'd probably end up with something like `sendv(Connection, messages...)`.